ang/civicase/case/details/file-tab/directives/case-details-file-tab.directive.js

```diff
@@ -9,17 +9,154 @@
         item: '=civicaseCaseDetailsFileTab',
         refresh: '=?refreshCallback'
       },
-      controller: civicaseCaseDetailsFileTabController
+      controller: 'civicaseCaseDetailsFileTabController'
     };
+  });
+
+  module.controller('civicaseCaseDetailsFileTabController', civicaseCaseDetailsFileTabController);
+
+  /**
+   * Controller function for the directive
+   *
+   * @param {object} $scope controllers scope object
+   * @param {object} BulkActions bulk actions service
+   * @param {object} crmApi service to call civicrm backend
+   * @param {Function} formatActivity format activity service
+   */
+  function civicaseCaseDetailsFileTabController ($scope, BulkActions, crmApi, formatActivity) {
+    $scope.ts = CRM.ts('civicase');
+    $scope.bulkAllowed = BulkActions.isAllowed();
+    $scope.isSelectAll = false;
+    $scope.isLoading = true;
+    $scope.selectedActivities = [];
+    $scope.totalCount = 0;
+    $scope.fileFilterParams = {
+      case_id: $scope.item.id,
+      text: '',
+      options: { xref: 1, limit: 0 }
+    };
+
+    $scope.findActivityById = findActivityById;
+    $scope.toggleSelected = toggleSelected;
+    $scope.refresh = refresh;
+    $scope.getActivities = getActivities;
+
+    (function init () {
+      getActivities();
+
+      $scope.$on('civicase::bulk-actions::bulk-selections', bulkSelectionsListener);
+    }());
 
     /**
-     * Controller function for the directive
+     * Bulk Selection Event Listener
      *
-     * @param {Object} $scope
+     * @param {object} event event
+     * @param {string} condition condition
      */
-    function civicaseCaseDetailsFileTabController ($scope, BulkActions) {
-      $scope.ts = CRM.ts('civicase');
-      $scope.bulkAllowed = BulkActions.isAllowed();
+    function bulkSelectionsListener (event, condition) {
+      if (condition === 'none') {
+        deselectAllActivities();
+      } else if (condition === 'visible') {
+        selectDisplayedActivities();
+      } else if (condition === 'all') {
+        selectEveryActivity();
+      }
     }
-  });
+
+    /**
+     * Deselection of all activities
+     */
+    function deselectAllActivities () {
+      $scope.isSelectAll = false;
+      $scope.selectedActivities = [];
+    }
+
+    /**
+     * Find activity by given ID
+     *
+     * @param {Array} searchIn - array of activities to search into
+     * @param {number/string} activityID activityID
+     * @returns {object} activity
+     */
+    function findActivityById (searchIn, activityID) {
+      return _.find(searchIn, { id: activityID });
+    }
+
+    /**
+     * Get List of Activities
+     */
+    function getActivities () {
```

Comment:
this should be `loadActivities` instead. It doesn't return anything. We should use `getAbc` when returning a value, even if it's a promise. We should use `loadAbc` when we store the value directly without returning it.